scripts/Packages/Package_Gullwing__QFP_SOIC_SO/ipc_gullwing_generator.py

```diff
@@ -155,6 +163,15 @@ def __createFootprintVariant(self, device_params, header, dimensions, with_therm
         size_y = dimensions['body_size_y'].nominal
 
         pincount = device_params['num_pins_x']*2 + device_params['num_pins_y']*2
+        
+        if 'hidden_pins' in device_params:
+            pincount_text = '{}-{}'.format(pincount - len(device_params['hidden_pins']), pincount)
+            pincount = device_params['num_pins_x']*2 + device_params['num_pins_y']*2 - len(device_params['hidden_pins'])
```

Comment:
Why would you use `pincount` and overwrite its value in the next line?

Reply:
Line 169 is the actual number of pins/pads present in the device, taking into account the hidden (removed) pins. So it's just subtracting the hidden pins from the pincount calculated above. Line 165 is the 'full' pin count used if there are no hidden or deleted pins. Yeah, no need to do the math again. You're saying why don't I take the value from line 165 and, if there are hidden pins, just subtract the number of hidden pins. I just missed that. Will fix.

Reply:
I think my problem here is that the variable `pincount` has different meanings just a few lines apart. How about something like this?

```python
        # pincount value and pincount text change if there are hidden or deleted pins 
        pincount_full = device_params['num_pins_x']*2 + device_params['num_pins_y']*2
        if 'hidden_pins' in device_params:
            pincount = pincount - len(device_params['hidden_pins'])
            pincount_text = '{}-{}'.format(pincount_full, pincount))
        elif 'deleted_pins' in device_params:
            pincount = pincount - len(device_params['deleted_pins'])
            pincount_text = '{}-{}'.format(pincount_full, pincount))
        else:
            pincount = pincount_full
            pincount_text = '{}'.format(pincount)
```

Reply:
Fine with me. I don't mind overwriting variables but that works too. Done.

I believe finding the pincount with hidden or deleted pins needs to subtract the length of the list from `pincount_full` so I made that correction from what is shown above. Let me know if I'm being a bonehead and my commit to address this isn't right.

I think it's too much to make a full document and/or rules out of it, considering the contributors writing scripts, but I wonder if there's a reason to gather up coding style preferences. I'd rather include someone's contribution if it works rather than turn them away or make it a burden if they're just learning and could have a more rewarding experience. And if the scripts, in general, are stocked with the desired coding style then it's easier to use them as a seed for starting a new script and get it reviewed and merged faster. (Additionally, I generally think all our scripts are dramatically under-commented but I usually follow along instead of adding the volume of comments my brain desires.)